### DISCUSSION TITLE
[A/L] (2021/22) Relazioni - Esercizio 19

### DISCUSSION THREAD

**Discussion by matypist** [sapienzastudentsnetwork/metodi-matematici-per-l-informatica#291] [Esercizi A-L]
![image](https://user-images.githubusercontent.com/69389461/152320297-fda6886a-8fa1-4aa3-973a-9918bc44bb5d.png)

**Comment by LuginiAndrea** [comment]
## Edit 
Mi sono accorto di non aver notato la parte o $y < w$. Fare riferimento quindi alla risposta sotto la mia.
-----
Userò la teoria degli insiemi in quanto un intervallo può essere visto come insieme (infinito) di tutti i valori presenti in esso
E' riflessiva: ogni intervallo è sottoinsieme largo di se stesso in quanto uguale a se stesso.
E' anti-simmetrica: Per teoria degli insiemi, se due insiemi sono sottoinsieme largo dell'altro a vicenda i due insiemi sono uguali
E' transitiva: Per teoria degli insiemi, $X \subseteq Y \subseteq Z \Longrightarrow X \subseteq Z$. Facile vedere come si applica agli intervalli

Vediamo come la relazione è un ordine parziale (non totale: pensiamo a $\left[1,2\right], \left[3,4\right]$)

**Comment by CiottoloMaggico** [comment]
![immagine](https://user-images.githubusercontent.com/79638284/216426860-59163820-5a9d-47da-b0ca-f448c315dd78.png)

**Comment by Exyss** [comment]
La dimostrazione di @CiottoloMaggico è **corretta ma incompleta** dal punto di vista dimostrativo, poiché non sono stati trattati i casi impossibili riguardo l'antisimmetria.

Il seguente commento è solo una formalizzazione della correttezza dell'intera dimostrazione.

------------

### Verifica Riflessività :heavy_check_mark: 

Per ogni intervallo si ha che $[x,y] \subseteq [x,y] \implies [x,y] \prec [x,y]$, dunque **la relazione è riflessiva** 

-----

### Verifica Antisimmetria :heavy_check_mark: 

Consideriamo $[x,y] \prec [w,z]$ e $[w,z] \prec [x,y]$. Preliminarmente, affermiamo che per definizione dell'intervallo $[x,y]$ si ha che $x \leq y$ e che per definizione dell'intervallo $[w,z]$ si ha che $w \leq z$.

* Se $[x,y] \prec [w,z]$ poiché $[x,y] \subseteq [w,z]$ mentre $[w,z] \prec [x,y]$ poiché $[w,z] \subseteq [x,y]$, allora
$[x,y] \subseteq [w,z] \land [w,z] \subseteq [x,y] \iff [x,y] = [w,z]$
dunque in tal caso l'antisimmetria $\underline{\text{persiste}}$

* Se $[x,y] \prec [w,z]$ poiché $[x,y] \subseteq [w,z]$ mentre $[w,z] \prec [x,y]$ poiché $z < x$, allora si ha che
$[x,y] \subseteq [w,z] \implies w \leq x \leq y \leq z$, contraddicendo l'ipotesi per cui $z < x$, dunque tale caso risulta essere $\underline{\text{impossibile}}$

* Se $[x,y] \prec [w,z]$ poiché $y < w$ mentre $[w,z] \prec [x,y]$ poiché $[w,z] \subseteq [x,y]$, allora si ha che
$[w,z] \subseteq [x,y] \implies x \leq w \leq z \leq y$, contraddicendo l'ipotesi per cui $y < w$, dunque tale caso risulta essere $\underline{\text{impossibile}}$

* Se $[x,y] \prec [w,z]$ poiché $y < w$ mentre $[w,z] \prec [x,y]$ poiché $z < x$, allora si ha che
$y < w \leq z < x$, contraddicendo la condizione di esistenza per cui $[x,y] \implies x < y$, dunque tale caso risulta essere $\underline{\text{impossibile}}$

Poiché solo il primo caso è possibile e poiché in tal caso l'antisimmetria persiste, allora **la relazione è antisimmetrica**

-------

### Verifica Transitività :no_entry_sign: 

Consideriamo $[x,y] \prec [w,z]$ e $[w,z] \prec [a,b]$. Preliminarmente, affermiamo che $[x,y] \implies x \leq y$, che $[w,z] \implies w \leq z$ e infine che $[a,b] \implies a \leq b$.

* Se $[x,y] \prec [w,z]$ poiché $[x,y] \subseteq [w,z]$ mentre $[w,z] \prec [a,b]$ poiché $[w,z] \subseteq [a,b]$, allora
$[x,y] \subseteq [w,z] \subseteq [a,b] \implies [x,y] \prec [a,b]$
dunque in tal caso la transitività  $\underline{\text{persiste}}$

* Se $[x,y] \prec [w,z]$ poiché $[x,y] \subseteq [w,z]$ mentre $[w,z] \prec [a,b]$ poiché $z < a$, allora si ha che
$[x,y] \subseteq [w,z] \implies w \leq x \leq y \leq z$, dunque $y \leq z < a \implies y < a \implies [x,y] \prec [a,b]$
dunque in tal caso la transitività $\underline{\text{persiste}}$

* Se $[x,y] \prec [w,z]$ poiché $y < w$ mentre $[w,z] \prec [a,b]$ poiché $[w,z] \subseteq [a,b]$, allora si ha che
$[w,z] \subseteq [a,b] \implies a \leq w \leq z \leq b$, andando a creare tre sottocasi:
    * Se $x \leq y \leq a$, allora $y \leq a \implies [x,y] \prec [a,b]$
    * Se $a \leq x \leq y$, allora $a \leq x \leq y < w \leq z \leq b \implies [x,y] \subseteq [a,b] \implies [x,y] \prec [a,b]$
    * Se $x \leq a \leq y$, allora $x \leq a \leq y < w \leq z \leq b \implies [x,y] \not\subseteq [a,b] \land y \nless a \implies [x,y] \not\prec [a,b]$
dunque in tal caso la transitività $\underline{\text{non persiste}}$

Poichè esiste almeno un caso in cui $[x,y] \prec [w,z] \land [w,z] \prec [a,b] \centernot\implies [x,y] \prec [a,b]$, allora **la relazione non è transitiva**

-------

EDIT: a voler essere precisi, anche il secondo sottocaso del terzo caso della transitività risulta invalidare la condizione necessaria, poiché $a \leq x \leq y < z \leq b \implies y \neq b$, dunque applicando la vera definizione di sottoinsieme proprio ed improprio si ha che $[x,y] \subset [a,b]$ mentre $[x,y] \not\subseteq [a,b]$. Quindi, possiamo concludere che anche in tale sottocaso la transitività non persista.
